scalding-core/src/main/scala/com/twitter/scalding/DescribedExecutionApp.scala

```diff
@@ -0,0 +1,57 @@
+package com.twitter.scalding
+
+sealed abstract class ArgOptions
+
+object RequiredArg extends ArgOptions
+object OptionalArg extends ArgOptions
+object ListArg extends ArgOptions
+object BooleanArg extends ArgOptions
+
+case class DescribedArg(key: String, description: String, argOption: ArgOptions)
+
+case class ArgHelp(describedArgs: Seq[DescribedArg]) {
+  def argString: String = {
+    describedArgs.foldLeft("") { case (str, DescribedArg(key, description, opt)) =>
+      val msg = opt match {
+        case RequiredArg => s"--$key VALUE "
+        case OptionalArg => s"[--$key VALUE] "
+        case ListArg => s"[--$key VALUE VALUE2] "
+        case BooleanArg => s"[--$key] "
+      }
+      str + msg
+    } + "[--help]"
+  }
+
+  def help: String = {
+    describedArgs.foldLeft("") { case (str, DescribedArg(key, description, opt)) =>
+      str + s"--$key($opt) :: $description \n"
+    } + "--help :: Show this help message."
+  }
+}
+
+/**
+  * Extending this Trait allows for descriptions to be added to Jobs
+  * If you run a job with --help the descriptions are output
+  */
+trait DescribedExecutionApp { this: ExecutionApp =>
```

Comment:
rather than a mixin style could we do:
```
object ArgHelp {
  def with(describedArgs: Seq[DescribedArg], ex: Execution[Unit]): Execution[Unit] = 
    // your code here?
}
```
Then it could be used with other Execution combinators.